src/Symfony/Component/String/Slugger/AsciiSlugger.php

```diff
@@ -113,10 +113,8 @@ public function slug(string $string, string $separator = '-', string $locale = n
             $transliterator = (array) $this->createTransliterator($locale);
         }
 
-        if (\is_string($this->emoji)) {
-            $transliterator[] = EmojiTransliterator::create("emoji-{$this->emoji}");
-        } elseif ($this->emoji && null !== $locale) {
-            $transliterator[] = EmojiTransliterator::create("emoji-{$locale}");
+        if ($emojiTransliterator = $this->createEmojiTransliterator(\is_string($this->emoji) ? $this->emoji : ($this->emoji ? $locale : null))) {
```

Comment:
I would pass the locale only to the fonction. And do some check there.
Ternary are harder to ready and I don't even talk about nested ternary